libvirt/tests/src/cpu/secure_host_validate.py

```
@@ -0,0 +1,101 @@
+        """
+        Verify secure launch prerequisites of the host.
+        """
+        def _validate_cpu_vendor():
+            exp_vendor = params.get("cpu_vendor")
+            act_vendor = cpu_utils.get_vendor()
+            if exp_vendor and exp_vendor != act_vendor:
+                test.cancel("%s is not supported on %s CPU." % (secure_feature, act_vendor))
+
+        def _validate_cpu_flag():
+            host_cpu_flags = set(cpu.get_cpu_flags())
+            cpu_flags = eval(params.get("cpu_flags", "[]"))
```

Ruff (0.15.2)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)